conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/RequestContext.java

```diff
@@ -44,4 +45,10 @@
      * Returns all query parameters associated with the current request.
      */
     ListMultimap<String, String> queryParameters();
+
+    /**
+     * Associates an {@link Arg} with the current request for observability.
+     * Implementations may choose to include this data in the request log.
+     */
+    void requestLogParameter(Arg<?> arg);
```

Comment:
Curious what we do when these are null? Defo not throw right?

Reply:
Yep, this is where things get a little funny though: I'd like the contract to be defined as non-null because the call isn't helpful in that case, however given an unexpected null input I'd expect the value to be ignored. I have a pretty strong stance against observability frameworks throwing because they're most utilized when assumptions have already been broken.

Reply:
yeah, log + metric on misuse is the way to go.